backend/src/jobs_server/utils/kueue.py

```diff
@@ -177,17 +177,54 @@ def pod(self) -> client.V1Pod:
 
         if self.owner_uid is None:
             self.owner_uid = self.managed_resource.metadata["uid"]
-        owner_uid = self.owner_uid
-        podlist: client.V1PodList = api.list_pod_for_all_namespaces(
-            label_selector=f"controller-uid={owner_uid}"
-        )
-        pods = podlist.items
+
+        if self.managed_resource.kind == "Job":
+            # Jobs are simple, we can use the `controller-uid` label to find the associated pods.
+
+            owner_uid = self.owner_uid
```

Comment:
I would inline this into the pod listing API call below, as it seems unused otherwise.